bastionzero/service/targets_disambiguated/all_targets.go

```diff
@@ -0,0 +1,132 @@
+)
+
+const (
+	targetsBasePath = "api/v2/targets"
+)
+
+// AllTargetsService handles communication with the disambiguated targets endpoints
+// of the BastionZero API.
+//
+// BastionZero API docs:
+//
+// - https://cloud.bastionzero.com/api
+
+type AllTargetsService client.Service
+
+type AccessDetails struct {
+	JIT                  bool            `json:"jit"`
+	AccessExpirationTime types.Timestamp `json:"accessExpirationTime"`
+}
+
+// Port describes a port number
+type Port struct {
+	Value *int `json:"value,omitempty"`
+}
+
+// TargetInterface lets you work with common attributes from any kind of
+// BastionZero target
+type TargetInterface interface {
+	// GetID returns the target's unique ID.
+	GetID() string
+	// GetName returns the target's name.
+	GetName() string
+	// GetStatus returns the target's status.
+	GetStatus() targetstatus.TargetStatus
+	// GetEnvironmentID returns the target's environment's ID.
+	GetEnvironmentID() string
+	// GetLastAgentUpdate returns the target's last agent update if set.
+	// Otherwise, returns nil.
+	GetLastAgentUpdate() *types.Timestamp
+	// GetAgentVersion returns the target's agent version.
+	GetAgentVersion() string
+	// GetRegion returns the target's region.
+	GetRegion() string
+	// GetAgentPublicKey returns the target's agent public key.
+	GetAgentPublicKey() string
+	// GetTargetType returns the target's type.
+	GetTargetType() targettype.TargetType
+}
+
+// Target abstracts common attributes from any kind of BastionZero target
+type Target struct {
+	// ID is the unique ID of the target
+	ID string `json:"id"`
+	// Name is the name of the target
+	Name string `json:"name"`
+	// Status is the condition of the target
+	Status targetstatus.TargetStatus `json:"status"`
+	// EnvironmentID is the ID of the environment the target belongs to
+	EnvironmentID string `json:"environmentId"`
+	// EnvironmentName is the name of the environment the target belongs to
+	EnvironmentName string `json:"environmentName"`
+	// Agent is the agent associated with this target
+	Agent *agents.AgentSummary `json:"agent"`
+	// AccessDetails describes how the target can be accessed
+	AccessDetails *AccessDetails `json:"accessDetails"`
+}
+
+func (t *Target) GetID() string                        { return t.ID }
+func (t *Target) GetName() string                      { return t.Name }
+func (t *Target) GetStatus() targetstatus.TargetStatus { return t.Status }
+func (t *Target) GetEnvironmentID() string             { return t.EnvironmentID }
+func (t *Target) GetEnvironmentName() string           { return t.EnvironmentName }
+func (t *Target) GetAgent() *agents.AgentSummary       { return t.Agent }
+func (t *Target) GetAccessDetails() *AccessDetails     { return t.AccessDetails }
+
+type ListAllTargetsOptions struct {
+	// Each target's list of connections will include connections with only these states. Defaults to only Open.
+	ConnectionStates connectionstate.ConnectionState `url:"connectionStates,omitempty"`
```

Comment:
```suggestion
	ConnectionStates []connectionstate.ConnectionState `url:"connectionStates,omitempty"`
```

I think this is a list on backend